posts/2021-08-01-disjoint-capture-in-closures.md

```diff
@@ -0,0 +1,62 @@
+    
+This is a basic closure that prints "Hello, World!":
+
+```rust
+let c = || println!("Hello, World!"); 
+```
+
+
+Closures can capture variables from the scope they are defined in.
+ ```rust
+let my_string = "Hello, World";
+let c = || println!("{}", my_string); // closure captures (& my_string)
+```
+
+
+If you're new to Rust and haven't used closures, we recommend you check out the [Rust Book](https://doc.rust-lang.org/book/ch13-01-closures.html) and [Rust by Example](https://doc.rust-lang.org/rust-by-example/fn/closures.html) to learn how to use closures.
+
+For those who would like a quick recap, we recommend reading the [Rust Reference](https://doc.rust-lang.org/reference/types/closure.html).
+
+
+## What is disjoint capture?
+
+"Disjoint capture" is the ability of a closure to capture only part of a variable rather than the variable in its entirety (the behavior in Rust 2018). The goal with this feature is to unify the semantics of closure captures with those of borrowing and move operations.
+
+Consider a `Point` `p` with fields `x` and `y` corresponding to the x and y coordinates of the point.
+
+```rust
+let p = Point { x: 10, y: 20 };
+
+let c = || println!("{}", p.x);
+```
+
+`c` in Rust 2018 will capture `p` completely, but in Rust 2021 it will only capture `p.x`. This now matches what gets borrowed when someone writes `println!("{}", p.x)`.
+
+Disjoint capture was proposed as part of [RFC 2229](https://github.com/rust-lang/rfcs/blob/master/text/2229-capture-disjoint-fields.md), and we suggest reading the RFC to better understand motivations behind the feature.
+
+The precise path that gets captured is typically the full path that is used in the closure, but there are cases where we will only capture a prefix of the path. See this pending PR to the [Rust Reference](https://github.com/rust-lang/reference/blob/6b88e48ffebc46be91884ef6237accb947e5f6f3/src/types/closure.md) for the full details.
+
+The feature also includes (minor) breaking changes to the Rust semantics which are also documented in this pending PR to the [Rust Reference](https://github.com/rust-lang/reference/blob/6b88e48ffebc46be91884ef6237accb947e5f6f3/src/types/closure.md).
+
+## How to use the feature?
+
+Interested in testing this out? You can now try disjoint capture in Rust closures on rust nightly using `#![feature(capture_disjoint_fields)]`. 
+
+If you would like to be warned of semantics change that may impact your code, you can follow migration instructions provided in the [2021 Edition Guide](https://doc.rust-lang.org/nightly/edition-guide/rust-2021/disjoint-capture-in-closures.html#migration).
+
+## How to submit bugs?
+
+To submit a bug simply [open an issue](https://github.com/rust-lang/rust/issues/new/choose) and tag the RFC 2229 working group using `@rust-lang/wg-rfc-2229`. We hope to create the best experince posible so no issue is too small, even a confusing error messages is worth reporting.
```

Comment:
Most folks won't be able to tag the working group (you need to be part of the org to ping teams). I think the easiest is to just ask folks to leave a label via rustbot, perhaps?

Reply:
How about we ask users to create issue directly on the 2229 repo? https://github.com/rust-lang/project-rfc-2229. MIght be a bit easier than trying to use rustbot

@nikomatsakis  what are your thoughts?

Reply:
Hmm! I think I'd prefer to have people file bugs on rust-lang/rust. I'm not sure what's the best way for people to tag disjoint closures though.